### ISSUE TITLE
MongoDB Queries

### ISSUE THREAD

**Issue by rkw6086** [TablePlus/TablePlus#1516]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
Mongo 3.6.12

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
Version 2.10.2 (273)

3. The steps to reproduce this issue:

When I am using MongoDB, I want to be able to write my own queries so that I can test queries I'm writing in my application. 

When I have the collection open, click the "SQL" tab. Even though SQL isn't supported by MongoDB, I would expect to at least be able to write the functions to query collections, like: 

`db.myData.find({isActive:true})`

Is there a way to write queries manually for MongoDB in TablePlus?

**Comment by huyphams** [comment]
TablePlus does not support mongo shell command at the moment, you can do it by using the advanced filter. Open the collection and press Command + F.

**Comment by ryan-idea42** [comment]
Is there a plan to support shell commands at some point in time? 

It would be a helpful feature for pro-users who are, like me, needing to quickly test out a query.

**Comment by huyphams** [comment]
Yes, we do have a plan to add mongo shell support in the future release but currently no deadline for it.

**Comment by ctrlaltdylan** [comment]
Honestly waiting for more mongo support before purchasing the full product. Would love to support but it's a deal breaker for me if it's not supported just a little bit better - including this feature request.

**Comment by roydondsouza** [comment]
Same, waiting for the mongo shell support. Everything else is great. 🍺 ❤️

**Comment by HuangJunjia** [comment]
I think it's important. I hope tableplus can add this feature in the future.❤️🎉

**Comment by nickpoulos** [comment]
Still no Mongo shell support??

**Comment by berkaey** [comment]
Ready to buy when mongo shell support is added!

**Comment by caojunkai01** [comment]
Ping, It's been a year, is there any progress on this issue?

**Comment by alii** [comment]
Any update? 😃

**Comment by woaiso** [comment]
Any update？

**Comment by brentgreeff** [comment]
I am using the filter option.
I entered - "user_id":"1234"
I know that 1234 exists, I can see it in the standard list view.
TP does not complain about the syntax - the bar is green - but I get no results.

**Comment by thexzan** [comment]
Any update？

**Comment by volnei** [comment]
Two years, a paid licence and no responses.. sad

**Comment by goryny4** [comment]
come on, guys! at least something?

**Comment by lewiuberg** [comment]
If it had a MongoDB shell, we would probably get it for the whole team. I like what I have seen so far 👍

**Comment by colwellkr** [comment]
I completely agree. Just discovered the app and the license cost is a little steep for personal use, but it'd be worth it if it supported mongo shell.

**Comment by lkobylski** [comment]
Any update here?

**Comment by cpuccino** [comment]
Still no updates? I actually bought table plus a while back but refunded cause of the lack of mongo shell support. Was kinda hoping it’s here now, but oh well, I’ll check back again in a couple months.

**Comment by mrcn04** [comment]
Any update?

**Comment by alexdonh** [comment]
+1 for mongo shell command support

**Comment by wickdninja** [comment]
+1 for mongo shell command support

**Comment by alexdonh** [comment]
just realized its almost 2 years now... perhaps mongo shell is not something TablePlus will have.

**Comment by ali-serin** [comment]
Wow, that was long. Still no update, ey. Sad.

**Comment by belgacea** [comment]
Won't pay a licence until it's a thing, sorry

**Comment by belgacea** [comment]
At least close the issue bro XD

**Comment by volnei** [comment]
This tool is terrible, and the only thing I expected was that it would support this feature, since they advertise with mongodb support. To complete yet a license that was lifetime now expires. I hope no one else buys it.

**Comment by bhumit070** [comment]
Hello @huyphams I know everyone wants this feature, and I think there might be some issues while implementing this feature from table plus, but it's been three years can you please add some light on the same like why are you not implementing it or any issues? and if you are working on it or not

**Comment by huyphams** [comment]
As far as I know, there is no implement of the MongoShell for C/C++ or Native. The MongoShell is a Javascript app.

**Comment by harisvsulaiman** [comment]
> As far as I know, there is no implement of the MongoShell for C/C++ or Native. The MongoShell is a Javascript app.

You can make the mongosh cli executable bundled with tableplus

**Comment by huyphams** [comment]
It won't/can't reuse the connection from the native driver.

**Comment by bhumit070** [comment]
> As far as I know, there is no implement of the MongoShell for C/C++ or Native. The MongoShell is a Javascript app.

Yes you are correct it in in js ,
So what is plan now, just asking I know you might already have done to integrate it in table plus to make it more awesome,
but just wante to know if this feature will be there or not in table plus?

**Comment by kakasoo** [comment]
I'm a long-time table plus user in RDB environment. I really need this function because I use NoSQL in my new company. Any update?

**Comment by huyphams** [comment]
It will support in the next version of the app (maybe next month).

**Comment by mrcn04** [comment]
Hopefully, it will, we have been waiting for literally years now 😄

**Comment by bhumit070** [comment]
Hello @huyphams  can you comment here once you release it so we can all be notifier here.

**Comment by huyphams** [comment]
Sure!

**Comment by huyphams** [comment]
I'm working on it, please reply to this email with the most frequently used command so we can prioritize it.
<img width="1166" alt="Screenshot 2023-09-29 at 8 35 26 AM" src="https://github.com/TablePlus/TablePlus/assets/3654622/3357277b-881d-49df-a909-d31be9ba85c5">

**Comment by unrelaxs** [comment]
Oh my god, the mongoShell is finally implemented. I'm so happy!

**Comment by unrelaxs** [comment]
run command:
`db.plm_pic_review_record.find({ 
  "created_at": { 
    "$gte": ISODate("2022-11-10T00:00:00.000Z"), 
    "$lt": ISODate("2022-11-11T00:00:00.000Z")
      }
})`

it doesnt work,and return error: Query 1 ERROR at Line 1: : Got parse error at "S", position 35: "SPECIAL_EXPECTED";

**Comment by huyphams** [comment]
I think the issue is ISODate, basically, MongoShell will replace it with a build in functions. We will implement it.

**Comment by unrelaxs** [comment]
> I think the issue is ISODate, basically, MongoShell will replace it with a build in functions. We will implement it.

Sure, I'm looking forward to it.

**Comment by DaveWilcock** [comment]
I just wanted to say thanks for this! Big improvement!

**Comment by yangg** [comment]
Failed
```
db.inventory.insert([
    { "_id" : 1, "sku" : "almonds", description: "product 1", "instock" : 120 },
    { "_id" : 2, "sku" : "bread", description: "product 2", "instock" : 80 },
    { "_id" : 3, "sku" : "cashews", description: "product 3", "instock" : 60 },
    { "_id" : 4, "sku" : "pecans", description: "product 4", "instock" : 70 },
    { "_id" : 5, "sku": null, description: "Incomplete" },
    { "_id" : 6 }
 ]);
```
Query 1 ERROR at Line 1: : Got parse error at "d", position 75: "SPECIAL_EXPECTED"

Version 5.6.2 (516)

**Comment by cesarvspr** [comment]
> Oh my god, the mongoShell is finally implemented. I'm so happy! @huyphams

Since what version?

**Comment by bmax** [comment]
> I think the issue is ISODate, basically, MongoShell will replace it with a build in functions. We will implement it.

Any way to use this without having to implement each built-in function?

**Comment by dogeweaver** [comment]
db.trading_position.aggregate([
    {
        $match: {
            "inst_id": "SWAP"
        }
    }
])

Query 1 ERROR at Line 1: : Command is not supported

What problem has been triggered? The same query can be executed normally in mongosh.

**Comment by HelenaShaw** [comment]
Thanks very much. Is there any documentation about what has been implemented so far? 
In particular, could you let me know if sorting is working? And if so, how to use it?

**Comment by fiko** [comment]
its not working for me, yet its working on mongosh:

```
db.users.updateOne(
	{"_id": ObjectId("6427f848b39a9135e7b7e63e")},
	{"$set" : {
			"value": "on"
	}}
)
```

**Comment by patrykk21** [comment]
can't query items with `ObjectId` - is there a guide or readme about table plus mongodb queries?

**Comment by huyphams** [comment]
Mongo Shell now available on the TablePlus Beta (Settings > Receive Beta Updates)

<img width="1255" height="991" alt="Image" src="https://github.com/user-attachments/assets/b9c92444-09f2-449c-bd3f-c1ccb1f955ec" />
